fiftyone/core/map/mapper.py

```diff
@@ -0,0 +1,254 @@
+              sample.
+            progress (Union[bool, Literal[&quot;workers&quot;]]): Whether or
+              not and how to render progress.
+            save (bool, optional): Whether to save mutated samples mutated in
+              the map function. Defaults to False.
+            skip_failures (bool, optional): Whether to gracefully continue
+              without raising an error if the map function raises an exception
+              for a sample. Defaults to True.
+
+        Yields:
+            Iterator[Tuple[bson.ObjectId, R]]: The sample ID and the result of
+              the map function for the sample.
+        """
+
+        if check_if_return_is_sample(sample_collection, map_fcn):
+            raise ValueError("`map_fcn` should not return Samples objects.")
+
+        yield from self._map_samples(
+            sample_collection,
+            map_fcn,
+            progress=progress,
+            save=save,
+            skip_failures=skip_failures,
+        )
+
+
+class LocalMapper(Mapper, abc.ABC):
+    """Base class for mapping samples in parallelizing on the same machine"""
+
+    @abc.abstractmethod
+    def _map_samples_multiple_workers(
+        self,
+        sample_collection: SampleCollection[T],
+        map_fcn: Callable[[T], R],
+        *_,
+        progress: Optional[Union[bool, Literal["workers"]]],
+        save: bool,
+        skip_failures: bool,
+    ) -> Iterator[
+        Tuple[bson.ObjectId, Union[Exception, None], Union[R, None]]
+    ]:
+        """Applies map function to each sample batch and returns an iterator
+          of the results.
+
+        Args:
+            sample_collection (SampleCollection[T]): The sample collection to
+              map.
+            map_fcn (Callable[[T], R]): The map function to apply to each
+              sample.
+            progress (Union[bool, Literal[&quot;workers&quot;]]): Whether or
+              not and how to render progress.
+            save (bool, optional): Whether to save mutated samples mutated in
+              the map function. Defaults to False.
+            skip_failures (bool, optional): Whether to gracefully continue
+              without raising an error if the map function raises an exception
+              for a sample. Defaults to True.
+
+        Yields:
+            MapSampleBatchesReturnType: The sample ID, the exception raised
+              (if any), and the result of the map function for the sample
+              (if no exception).
+        """
+
+    def _map_samples(
+        self,
+        sample_collection: SampleCollection[T],
+        map_fcn: Callable[[T], R],
+        *_,
+        progress: Optional[Union[bool, Literal["workers"]]] = None,
+        save: bool = False,
+        skip_failures: bool = True,
+    ) -> Iterator[Tuple[bson.ObjectId, R]]:
+        # If workers if 1 on a the same local machine, no need for the
+        # overhead of trying to parallelize.
```

Comment:
grammatically incorrect, also not sure what it's trying to say about same local machine.

Reply:
Addressed here: https://github.com/voxel51/fiftyone/pull/5653/files